packages/appkit/src/client.ts

```diff
@@ -575,40 +644,667 @@ export class AppKit {
+      })
+
+      if (res) {
+        this.syncProvider(res)
+        await this.syncAccount(res)
+      }
+    }
+  }
+
+  private getAdapter(namespace: ChainNamespace) {
+    return this.chainAdapters?.[namespace]
+  }
+
+  private createUniversalProvider() {
+    if (
+      !this.universalProviderInitPromise &&
+      typeof window !== 'undefined' &&
+      this.options?.projectId
+    ) {
+      this.universalProviderInitPromise = this.initializeUniversalAdapter()
+    }
+
+    return this.universalProviderInitPromise
+  }
+
+  private async initializeUniversalAdapter() {
+    const universalProviderOptions: UniversalProviderOpts = {
+      projectId: this.options?.projectId,
+      metadata: {
+        name: this.options?.metadata ? this.options?.metadata.name : '',
+        description: this.options?.metadata ? this.options?.metadata.description : '',
+        url: this.options?.metadata ? this.options?.metadata.url : '',
+        icons: this.options?.metadata ? this.options?.metadata.icons : ['']
+      }
+    }
+
+    this.universalProvider = await UniversalProvider.init(universalProviderOptions)
+  }
+
+  public async getUniversalProvider() {
+    if (!this.universalProvider) {
+      try {
+        await this.createUniversalProvider()
+      } catch (error) {
+        throw new Error('AppKit:getUniversalProvider - Cannot create provider')
+      }
+    }
+
+    return this.universalProvider
+  }
+
+  private createAuthProvider() {
+    const emailEnabled =
+      this.options?.features?.email === undefined
+        ? CoreConstantsUtil.DEFAULT_FEATURES.email
+        : this.options?.features?.email
+    const socialsEnabled = this.options?.features?.socials
+      ? this.options?.features?.socials?.length > 0
+      : CoreConstantsUtil.DEFAULT_FEATURES.socials
+    if (this.options?.projectId && (emailEnabled || socialsEnabled)) {
+      this.authProvider = W3mFrameProviderSingleton.getInstance({
+        projectId: this.options.projectId
+      })
+      this.listenAuthConnector(this.authProvider)
+    }
+  }
+
+  private async createAdapters(blueprints?: AdapterBlueprint[]): Promise<Adapters> {
+    if (!this.universalProvider) {
+      this.universalProvider = await this.getUniversalProvider()
+    }
+
+    this.syncRequestedNetworks()
+
+    return this.chainNamespaces.reduce<Adapters>((adapters, namespace) => {
+      const blueprint = blueprints?.find(b => b.namespace === namespace)
+
+      if (blueprint) {
+        adapters[namespace] = blueprint
+        adapters[namespace].namespace = namespace
+        adapters[namespace].construct({
+          namespace,
+          projectId: this.options?.projectId,
+          networks: this.caipNetworks
+        })
+        if (this.universalProvider) {
+          adapters[namespace].setUniversalProvider(this.universalProvider)
+        }
+        if (this.authProvider) {
+          adapters[namespace].setAuthProvider(this.authProvider)
```

Comment:
could it happen that authprovider is not suported on one of these blueprints / networks ?

Reply:
It could be, but the only thing is that we set the underlying frameProvider for the adapter. What the adapter does with this provider, can be defined in the client code